### PR TITLE
Move sections to a TableData class instead of duplicating the array

### PR DESCRIPTION
**Pull request by PierreOleo**
In the current implementation, when the user manually moves a row `tableView(_ tableView: UITableView, moveRowAt sourceIndexPath: IndexPath, to destinationIndexPath: IndexPath)` will modify the datasource `sections` but not the `sections` of `FunctionalTableData`. 

If the TableSection `didMoveRow` calls for a re-render then `FunctionalTableData` will detect changes. If the changes detected are "move" changes in an animated render, another unexpected move could be applied on top of the current one

### CODE REVIEW

File: FunctionalTableDataDemo/View Controllers/TableExampleController.swift [hunk 1]
```diff
@@ -11,28 +11,25 @@ import FunctionalTableData
...
-		items.append(NSDate().description)
+		items.append("\(items.count)")
+		render()
 	}
 	
 	private func render() {
 		let rows: [CellConfigType] = items.enumerated().map { index, item in
 			return LabelCell(
-				key: "id-\(index)",
+				key: "id-\(item)",
```
Comment (dannbeaus) [inline FunctionalTableDataDemo/View Controllers/TableExampleController.swift]: why this change? index helps with uniquing, I don't suggest making that change.

Comment (PierreOleo) [reply FunctionalTableDataDemo/View Controllers/TableExampleController.swift]: Will remove this change

The intention was to have a manual sorting simple sample code with moves detection. Keeping the index as an id makes keys vary for each object on each render when reordering. FunctionalTableData then detects an insertion and a deletion instead of a move.
Uniquing was possible as there is no way to delete a item from the array.

But not sure of the added value anymore, moreover with github.com/Shopify/FunctionalTableData/pull/147  removing this class really soon.

Comment (raulriera) [reply FunctionalTableDataDemo/View Controllers/TableExampleController.swift]: With the item it will crash if we spam the button right? But I agree with not using index here. I don't think it matters. I want to delete these examples anyway

**Comment by raulriera** [comment]
You had a previous implementation, what happened?

**Comment by PierreOleo** [comment]
@g-Off suggested the `TableData` approach as it would be cleaner than just passing arrays everywhere

File: FunctionalTableDataDemo/View Controllers/TableExampleController.swift (hunk 1 above)
Comment (PierreOleo) [reply FunctionalTableDataDemo/View Controllers/TableExampleController.swift]: Should not have crashed as `NSDate().description` was replaced by `items.count` but decided to revert the example changes

File: FunctionalTableData/TableView/FunctionalTableData.swift [hunk 2]
```diff
@@ -41,17 +41,18 @@ public class FunctionalTableData {
 	
 	private func dumpDebugInfoForChanges(_ changes: TableSectionChangeSet, previousSections: [TableSection], visibleIndexPaths: [IndexPath], exceptionReason: String?, exceptionUserInfo: [AnyHashable: Any]?) {
 		guard let exceptionHandler = FunctionalTableData.exceptionHandler else { return }
-		let exception = Exception(name: name, newSections: sections, oldSections: previousSections, changes: changes, visible: visibleIndexPaths, viewFrame: tableView?.frame ?? .zero, reason: exceptionReason, userInfo: exceptionUserInfo)
+		let exception = Exception(name: name, newSections: data.sections, oldSections: previousSections, changes: changes, visible: visibleIndexPaths, viewFrame: tableView?.frame ?? .zero, reason: exceptionReason, userInfo: exceptionUserInfo)
 		exceptionHandler.handle(exception: exception)
 	}
 	
-	private var sections: [TableSection] = [] {
+	private var data: TableData {
```
Comment (g-Off) [inline FunctionalTableData/TableView/FunctionalTableData.swift]: This can be a let now can't it? Should also be able to get rid of the didSet portion completely as well
